src/hera/workflows/_meta_mixins.py

```diff
@@ -427,10 +425,10 @@ def __call__(self, *args, **kwargs) -> Union[None, Step, Task]:
                     f"Callable Template '{self.name}' is not callable under a Workflow"  # type: ignore
                 )
             if isinstance(_context.pieces[-1], (Steps, Parallel)):
-                return Step(template=self, **kwargs)
+                return Step(template=cast(Templatable, self), **kwargs)
```

Comment:
should this class maybe extend Templatable?

Reply:
I originally wanted to have `TemplateMixin` itself extend `Templatable`, but `Protocol` seems incompatible with `BaseModel`:

```
TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
```